Lib/test/test_inspect/test_inspect.py

```diff
@@ -4622,6 +4622,122 @@ class D2(D1):
 
         self.assertEqual(inspect.signature(D2), inspect.signature(D1))
 
+    def test_signature_skip_bound_arg_method(self):
+        def decorator(func):
+            @functools.wraps(func)  # set `__wrapper__` attribute
+            def wrapper(*args, **kwargs):
+                return func(*args, **kwargs)
+            return wrapper
+
+        class My:
+            def method(self, arg: int) -> None: ...
+            @decorator
+            def decorated(self, arg: int) -> None: ...
+            @classmethod
+            def cl(cls, arg2: str) -> None: ...
+            @staticmethod
+            def st(arg1: str) -> bool: ...
+
+        for follow_wrapped in (True, False):
```

Comment:
This function has too many nested loops for my little brain :-( Can you rename `test_signature_skip_bound_arg_method` to `check_signature_skip_bound_arg_method` and have two methods:

```
def test_method_skip_bound_arg_follow_wrapped(self):
    self.check_signature_skip_bound_arg_method(follow_wrapped=True)

def test_method_skip_bound_arg_no_follow_wrapped(self):
    self.check_signature_skip_bound_arg_method(follow_wrapped=True)
```

Well. Something like that.